src/lib/assetBridger/l1l3Bridger.ts

```diff
@@ -0,0 +1,1166 @@
+
+  // todo: tune these
+  public readonly defaultRetryableGasParams: ManualRetryableGasParams = {
+    l2ForwarderFactoryGasLimit: BigNumber.from(1_000_000),
+    l1l2TokenBridgeGasLimit: BigNumber.from(1_000_000),
+    l2l3TokenBridgeGasLimit: BigNumber.from(1_000_000),
+    l1l2TokenBridgeRetryableSize: BigNumber.from(1000),
+    l2l3TokenBridgeRetryableSize: BigNumber.from(1000),
+  } as const
+
+  public constructor(public readonly l3Network: L2Network) {
+    super(l3Network)
+
+    if (!this.l2Network.teleporterAddresses) {
+      throw new ArbSdkError(
+        `L2 network ${this.l2Network.name} does not have teleporter contracts`
+      )
+    }
+
+    this.teleporterAddresses = this.l2Network.teleporterAddresses
+  }
+
+  /**
+   * Get the corresponding L2 token address for the provided L1 token
+   */
+  public async getL2ERC20Address(
+    erc20L1Address: string,
+    l1Provider: Provider
+  ): Promise<string> {
+    await this._checkL1Network(l1Provider)
+    return this._getChildErc20Address(
+      erc20L1Address,
+      l1Provider,
+      this.l2Network
+    )
+  }
+
+  /**
+   * Get the corresponding L3 token address for the provided L1 token
+   */
+  public async getL3ERC20Address(
+    erc20L1Address: string,
+    l1Provider: Provider,
+    l2Provider: Provider
+  ): Promise<string> {
+    await this._checkL2Network(l2Provider)
+    return this._getChildErc20Address(
+      await this.getL2ERC20Address(erc20L1Address, l1Provider),
+      l2Provider,
+      this.l3Network
+    )
+  }
+
+  /**
+   * Given an erc20 address on a parent network, get the corresponding erc20 address on a child network
+   */
+  private _getChildErc20Address(
+    erc20ParentAddress: string,
+    parentProvider: Provider,
+    childNetwork: L2Network
+  ) {
+    // assume that provider has been checked
+    const parentGatewayRouter = L1GatewayRouter__factory.connect(
+      childNetwork.tokenBridge.l1GatewayRouter,
+      parentProvider
+    )
+
+    return parentGatewayRouter.calculateL2TokenAddress(erc20ParentAddress)
+  }
+
+  /**
+   * Get the address of the L1 <-> L2 gateway on L1 given an L1 token address
+   */
+  public async getL1L2GatewayAddress(
+    erc20L1Address: string,
+    l1Provider: Provider
+  ): Promise<string> {
+    await this._checkL1Network(l1Provider)
+
+    return await L1GatewayRouter__factory.connect(
+      this.l2Network.tokenBridge.l1GatewayRouter,
+      l1Provider
+    ).getGateway(erc20L1Address)
+  }
+
+  /**
+   * Get the address of the L2 <-> L3 gateway on L2 given an L1 token address
+   */
+  public async getL2L3GatewayAddress(
+    erc20L1Address: string,
+    l1Provider: Provider,
+    l2Provider: Provider
+  ): Promise<string> {
+    await this._checkL1Network(l1Provider)
+    await this._checkL2Network(l2Provider)
+
+    const l2Token = await this.getL2ERC20Address(erc20L1Address, l1Provider)
+
+    return await L1GatewayRouter__factory.connect(
+      this.l3Network.tokenBridge.l1GatewayRouter,
```

Comment:
so l1GateRouter is really on L2 here, wondering if a clarifying comment would help? (might just be worse tho lol)